pkg/providers/vsphere/network/network.go

```diff
@@ -660,22 +660,27 @@ func vpcSubnetPortToResult(
 	}
 
 	ipConfigs := []NetworkInterfaceIPConfig{}
+	ipAddresses := subnetPort.Status.NetworkInterfaceConfig.IPAddresses
+
+	// For now, we have an assumption that one subnetport only have one IP address.
+	// An empty ipAddress is NSX Operator's way of saying DHCP.
+	if len(ipAddresses) != 0 && len(ipAddresses[0].IPAddress) != 0 {
```

Comment:
I think the `len(ipAddresses[0].IPAddress) != 0` part of this can be removed because it will hit the `continue` in the loop